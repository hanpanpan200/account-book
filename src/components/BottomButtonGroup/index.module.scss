@import "../../styles/colors";

.container {
  height: 60px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-sizing: content-box;
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: $white;

  .button {
    width: 48%;
    height: 40px;
    align-items: center;
    border: 1px solid #DBDBDB;
    border-radius: 3px;
    text-align: center;
    line-height: 40px;
  }
  .cancel {
    color: $black;
    background-color: $white;
  }
  .confirm {
    color: $white;
    background-color: $red;
  }
}