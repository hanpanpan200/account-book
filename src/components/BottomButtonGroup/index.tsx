import React from 'react';
import classNames from 'classnames';

import styles from './index.module.scss';

const cx = classNames.bind(styles);

interface Props {
  onCancel: () => void
  onConfirm: () => void
}

const BottomButtonGroup: React.FC<Props> = ({ onConfirm, onCancel }) => {
  return (
    <div className={styles.container}>
      <div className={cx(styles.button, styles.cancel)} onClick={onCancel}>取消</div>
      <div className={cx(styles.button, styles.confirm)} onClick={onConfirm}>确定</div>
    </div>
  )
};

export default BottomButtonGroup;
