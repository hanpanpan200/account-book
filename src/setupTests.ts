// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { fireEvent } from '@testing-library/react';

export const fireClickEvent = (element: Document | Element | Window | Node) => {
  fireEvent(element, new MouseEvent('click', {
    bubbles: true,
    cancelable: true
  }));
}