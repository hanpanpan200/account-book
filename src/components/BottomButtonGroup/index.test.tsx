import React from 'react';
import { render } from '@testing-library/react';
import { fireClickEvent } from 'setupTests';
import BottomButtonGroup from './index';

describe('BottomButtonGroup', () => {
  const mockedOnCancel = jest.fn();
  const mockedOnConfirm = jest.fn();

  it('should render correctly', () => {
    const { container } = render(<BottomButtonGroup onCancel={mockedOnCancel} onConfirm={mockedOnConfirm} />);
    expect(container).toMatchSnapshot();
  });

  it('should trigger onCancel and onConfirm successfully when buttons are clicked', () => {
    const { getByText } = render(<BottomButtonGroup onCancel={mockedOnCancel} onConfirm={mockedOnConfirm} />);
    const cancelButton = getByText('取消');
    const confirmButton = getByText('确定');
    fireClickEvent(cancelButton);
    expect(mockedOnCancel).toBeCalled();
    fireClickEvent(confirmButton);
    expect(mockedOnConfirm).toBeCalled();
  });
});

