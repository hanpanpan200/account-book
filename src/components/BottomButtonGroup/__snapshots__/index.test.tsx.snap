// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomButtonGroup should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="button cancel"
    >
      取消
    </div>
    <div
      class="button confirm"
    >
      确定
    </div>
  </div>
</div>
`;
